kernels/volk/volk_16i_max_star_horizontal_16i.h

```diff
@@ -128,7 +127,7 @@ volk_16i_max_star_horizontal_16i_a_ssse3(int16_t* target, int16_t* src0, unsigne
     p_target += 1;
   }
 
-  for(i = 0; i < intermediate; ++i) {
+  if (num_bytes >> 4 & 1) {
```

Comment:
Please leave the line `  int intermediate = (num_bytes >> 4) & 1;` and then use `intermediate`. Make it clear the connection between all of these statements. My request here is purely because this setting is used multiple times. For those other changes where the setting is used just once, it makes sense to remove the main variable and just use the setting directly in the code.

Reply:
No, in this case code vs. memory. Use the variable if it is more two times.

Reply:
Yes, it is bitwise simple, but, again, it's used 3 times. What I mean here is to change this line from the `for` to your new `if`, but do `if(intermediate) {` ... ditto the 2 parts below. That way, one can look at the code and see `intermediate` and note how it is used in 3 places, rather than having to parse `(num_bytes >> 4 & 1)` 3 times to determine the connection.

Reply:
I think we do not need to add that variable, compiler will optimise, so out of 3 manipulations it will use only 2. Memory allocations and using it from and in it, on the other hand... I suppose most of kernel should be done in AVX/2 registers, right?

Reply:
My ask here is to help keep the code as readable and interpretable as one reasonably can. Keeping those 3 uses under the same variable might or not be faster that using the setting directly; I don't know & am not going to go testing it. That said, keeping those 3 uses under the same variable definitely helps keep the code as readable and interpretable. Hence my request to revert just that 1 variable's changes.

Reply:
Okay, will try to fix all of those.